### ISSUE TITLE
ci.yaml check failing on Engine -> Framework roll

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#116989]
https://github.com/flutter/flutter/pull/116988

I am going to close it to trigger a new roll, but filing this so that it can be investigated.

**Comment by zanderso** [comment]
The next one also failed. Escalating to P0. https://github.com/flutter/flutter/pull/116990

**Comment by zanderso** [comment]
Looks like a mirroring issue:
```
HTTP 404: Not found: HTTP 404: https://flutter.googlesource.com/mirrors/flutter/+/c420f29838d82b451f4968e699c8985f270fc032/.ci.yaml?format=text
```

**Comment by sealesj** [comment]
Trying the fix mentioned here https://github.com/flutter/flutter/issues/101520 in order to resolve this

**Comment by godofredoc** [comment]
This was a transient error caused by the github CDN problem, several rolls and change have landed correctly since this morning.

The pending work is related to migrate to GitHub APIs with better SLOs.

**Comment by godofredoc** [comment]
Fix is work in progress.

**Comment by sealesj** [comment]
Since this is a one-off issue, decreasing the priority. The work is in progress, however requires significant testing changes

**Comment by ricardoamador** [comment]
I have doubts that this addresses the underlying issue. We are simply trading a call to the raw file via a commit ref for a call to the github api via a commit ref. If the commit ref does not exist then in either case the same outcome is possible. In both cases we fall back to the git on borg fetch which also seems odd since github is the source of truth here.

**Comment by ricardoamador** [comment]
I have added significantly more logging to determine where there are holes in our code and if we are failing silently around creation of the ci.yaml validation check.

**Comment by ricardoamador** [comment]
We need to revisit the back off logic and the wait time.

**Comment by ricardoamador** [comment]
It looks like this IS timing related. When this happens it looks like we are not giving enough time in our retries. From logging I can see that we go to get the ci.yaml for the current PR and it does not exist. However from the logs I can confirm that it does exists just that when we attempted 3 times to get the ci.yaml we did it within 1 second which is far too fast. I am going to submit a fix for the retryOptions to give it far more time to find the file.
![image](https://user-images.githubusercontent.com/32242716/216482556-440f0b47-b933-4ab2-96a9-c97472b48e61.png)
